opentracing-api/src/main/java/io/opentracing/Tracer.java

```diff
@@ -186,7 +198,7 @@
          * @see ActiveSpanSource
          * @see ActiveSpan
          */
-        ActiveSpan startActive();
+        Scope startActive();
```

Comment:
I was thinking; maybe with a decent observer definition, could it be feasible to let the ScopeManager implementation decide whether to auto-activate started spans?

If it can register itself as observer to started spans, it can make the decision to auto-activate or not (maybe even based on some inspection of the started span).

That way for the instrumentation the tracer API basically becomes simple again (pre-0.30), while affording adding such complexity if desired in the scope manager as implementation decision.

Just thoughts basically, would have to play with things to test feasibility...

Reply:
I suppose activation might be a decision of specific framework instrumentation that will not be aware if auto-activation is enabled or not.